spk/ffmpeg4/Makefile

```diff
@@ -0,0 +1,36 @@
+SPK_NAME = ffmpeg
+SPK_VERS = 4.4.3
+SPK_REV = 48
+SPK_ICON = src/ffmpeg.png
+CHANGELOG = "1. Change package directory to ffmpeg4<br/>2. Enable openh264<br/>3. Enable twolane<br/>4. Enable aarch64 asm neon optimization<br/>5. Update dav1d to version 1.1.0<br/>6. Update frei0r to version 2.2.0<br/>7. Update libvpx to version 1.13<br/>8. Update x264 to latest git commit<br/>9. Update SVT-AV1 to version 1.4.1<br/>10. Update libass to version 0.17.1"
+
+CONF_DIR = src/conf
+
+DEPENDS = cross/ffmpeg4
+
+# Enforce using ffmpeg4 directory
+INSTALL_PREFIX = /var/packages/ffmpeg4/target
```

Comment:
@hgy59 so I followed-up your advice and kept the `ffmpeg` naming for ffmpeg4.  While builds and everything works out all right it hapens that it ends-up getting installed in `/var/packages/ffmpeg/target`, hence not using the `ffmpeg4` directory name, hence generating an error at usage:
```
$ ./ffmpeg
./ffmpeg: error while loading shared libraries: libavdevice.so.58: cannot open shared object file: No such file or directory
```
The obvious fix is to either use the default directory name of `ffmpeg` or rename to `ffmpeg4`.... 

Or the best outcome, do you know if there a way to enforce the target installation directory name at package install?

Reply:
@th0ma7 first you have to check, whether there is a wrong library rpath in ffmpeg executable
```
# readelf -d /var/packages/ffmpeg/target/bin/ffmpeg

Dynamic section at offset 0x3c408 contains 36 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libavdevice.so.58]
 0x0000000000000001 (NEEDED)             Shared library: [libavfilter.so.7]
 0x0000000000000001 (NEEDED)             Shared library: [libavformat.so.58]
 0x0000000000000001 (NEEDED)             Shared library: [libavcodec.so.58]
 0x0000000000000001 (NEEDED)             Shared library: [libavresample.so.4]
 0x0000000000000001 (NEEDED)             Shared library: [libpostproc.so.55]
 0x0000000000000001 (NEEDED)             Shared library: [libswresample.so.3]
 0x0000000000000001 (NEEDED)             Shared library: [libswscale.so.5]
 0x0000000000000001 (NEEDED)             Shared library: [libavutil.so.56]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x000000000000000f (RPATH)              Library rpath: [/var/packages/ffmpeg/target/lib]
```

If the RPATH is wrong (i.e. uses ffmpeg4 instead of ffmpeg) we probably have an error in the `install_correct_lib_files` target of spksrc.install.mk.

If there is no common solution to fix this, you can use patchelf in a custom POST_STRIP_TARGET in spk/ffmpeg/Makefile.
As ffmpeg is not a service, it is not possible to add LD_LIBRARY_PATH for all use cases. So it is better to patch the ffmpeg executable with the correct RPATH (you can find an example this in spk/dotnet-runtime/Makefile).

Reply:
BTW the solution above is at build time, not at instsallation time.

Reply:
And that is where the issue is: I'm building ffmpeg4 so the destination directy ends-up being `ffmpeg4` ... But at installation time it gets installed into `ffmpeg` which matches the package name.  So yeah, the rpath is wrong once installed there for sure.

Question is, how can I change the installation directory as well?  I've looked in Syno dev guide and found nothing obvious so far.

Reply:
I don't know what's going wrong with the build, but the install directory (INSTALL_PREFIX et al.) must use `SPK_NAME` that is actually `ffmpeg` and not `ffmpeg4`.

Reply:
Exacly.  I've looked into the synology guide and there is no way to change that behavior (e.g. point towards another installation directory).  I've also looked into creating a symlink betwen `/var/packages/ffmpeg` and `/var/packages/ffmpeg4` but that won't work for DSM7 as requiring escalated privileges.

As such I've reverted `ffmpeg4` to be using both the `ffmpeg` name and using it's default installation directory name.  This also led me to readjust the calls for reusing libraries within comskip, chromaprint & tvheadend so the ability to easily switch from one version of ffmpeg to the next is maintained while managing the version 4 exception.

Next build should be now OK